### ISSUE TITLE
[Crash] RecreatableDeviceManager.OnAsynchronousCreateResourcesCompleted crashes with Access Violation when resource loading completed when Canvas is already unloaded.

### ISSUE THREAD

**Issue by mstyura** [Microsoft/Win2D#486]
RecreatableDeviceManager.OnAsynchronousCreateResourcesCompleted might be called when object is already destroyed. Canvas can be unloaded before async operation with resource loading completed.

```
HRESULT OnAsynchronousCreateResourcesCompleted(IAsyncAction*, AsyncStatus)
{
    return ExceptionBoundary(
        [&]
        {
            std::unique_lock<std::recursive_mutex> lock(m_currentOperationMutex); // <- crashes here when trying to acquire lock on this->m_currentOperationMutex
            m_currentOperationIsPending = false;
            lock.unlock();
            
            if (m_changedCallback)
                m_changedCallback(ChangeReason::Other);
        });
}

```

**Comment by shawnhar** [comment]
Thanks for the report, we'll take a look.

Do you have a reliable way to repro this crash?

**Comment by mstyura** [comment]
@shawnhar yes I have, but inside of existing app ([Viber](https://www.microsoft.com/en-us/store/p/viber/9wzdncrfj46z)). I'll try to provide simple app with repro.

**Comment by CMay** [comment]
Any update on this?

**Comment by shawnhar** [comment]
I haven't had a chance to look at this yet, but we're planning to dig into it for the next Win2D update (probably ~1 month out)

**Comment by shawnhar** [comment]
Hi,

I am working on a fix for this bug now, but the link to your repro is no longer valid.  Do you still have that project?

It would be useful to confirm that the fix I am making really does address the same issue you ran into :-)

**Comment by shawnhar** [comment]
Hmm, I've been attempting to modify SimpleSample to make it unload the control while an async resource load is in progress, but am not having any luck getting this to crash, and am running out of ideas to try :-(  Hopefully you still have access to the repro app!

**Comment by mstyura** [comment]
@shawnhar here is fixed link https://www.dropbox.com/s/2ujs4khwyon4rga/Win2DCrashRepro.zip?dl=0

**Comment by shawnhar** [comment]
Thanks!  I can repro the crash now.

**Comment by shawnhar** [comment]
This is fixed in Win2D 1.21.0 (released today)
